src/TestInfrastructure/Framework/TestLogger.cs

```diff
@@ -72,11 +87,15 @@ public void Reset()
         public void WriteLine(string message)
         {
             outputStrings.Add(message + Environment.NewLine);
+            if (logToConsole)
+            {
+                Console.WriteLine(message);
+            }
         }
 
         public void WriteLine(string messageFormat, params object[] args)
         {
-            outputStrings.Add(string.Format(System.Globalization.CultureInfo.CurrentCulture, messageFormat, args) + Environment.NewLine);
+            WriteLine(string.Format(System.Globalization.CultureInfo.CurrentCulture, messageFormat, args));
```

Comment:
shouldn't this check `logToConsole` value before?

if not, then why is it checked at line 90?

Reply:
This overload with parameters just formats the message and then calls the other overload, which actually does the recording and writing to the console. `logToConsole` doesn't need to be checked in both places.